### ISSUE TITLE
Clarification on reproduction settings: P2018 subset, loss weights, label smoothing, and batch size

### ISSUE THREAD

**Issue by seoyeonmmn** [wjq-learning/SleepDG#6]
Hello, thank you for releasing the code for SleepDG.

I am currently trying to reproduce the reported five-domain DG results using the official repository, and I found a few differences between the settings described in the paper and those in the released code. I would really appreciate some clarification on which settings were used to obtain the reported results.

### 1. P2018: first 150 recordings

In the paper, both SHHS and P2018 are described as using the **first 150 recordings** to balance the dataset sizes across domains.

For SHHS, this is explicitly reflected in the preprocessing code:

```python
for psg_f_name, label_f_name in tqdm(psg_label_f_pairs[:150]):
```

However, in `prepare_p2018.py`, all sorted subjects appear to be processed:

```python
subject_dirs = os.listdir(dir_path)
subject_dirs.sort()

...

for psg_f_name, label_f_name in tqdm(psg_label_f_pairs):
```

For reproducing the reported results, should P2018 also be restricted to the first 150 subjects after `subject_dirs.sort()`?

If possible, could you also confirm the exact selection rule or the subject IDs used for the 150 P2018 recordings?

### 2. Loss weights

The paper states that the coefficients for the reconstruction and domain alignment losses are:

```text
λ1 = λ2 = λ3 = 0.5
```

However, the released `trainer.py` uses:

```python
loss = loss1 + loss2 + loss3
```

where `loss2` is the CORAL-based alignment loss and `loss3` is the reconstruction loss.

For reproducing the reported results, should the implementation instead be:

```python
loss = loss1 + 0.5 * loss2 + 0.5 * loss3
```

assuming that `loss2` already contains both the epoch-level and sequence-level alignment losses?

Or were all three terms actually used with weight 1.0 for the reported experiments?

### 3. Label smoothing

The released `main_5.py` sets:

```python
--label_smoothing 0.1
```

and this value is passed to `CrossEntropyLoss`.

However, I could not find label smoothing mentioned in the experimental settings or classification loss formulation in the paper.

Was `label_smoothing=0.1` used for the reported results, or should standard cross-entropy (`label_smoothing=0`) be used for reproduction?

### 4. Batch size

The paper reports a batch size of **32**, while the released `main_5.py` uses:

```python
--batch_size 64
```

Could you confirm whether the reported results were obtained with batch size 32 or 64?

For reference, after using the paper-specified **batch size 32** and **loss weights 0.5**, while keeping the other released-code settings unchanged, I obtained:

```text
                ACC      Macro-F1
Sleep-EDFx      77.07      71.75
HMC             72.65      70.47
ISRUC           77.33      73.58
SHHS            72.38      65.24
P2018           73.92      69.31

Average         74.67      70.07
```

These results are reasonably close to the reported average performance, but I would like to make sure that I am using the exact experimental protocol before drawing conclusions about reproducibility.

Thank you very much for your time and for making the implementation publicly available.
